articles/css-grid/index.md

```diff
@@ -0,0 +1,379 @@
+
+A grid cell is the most elementary unit of a grid. It is made up of two adjacent row and column grid lines.
+
+### grid-track
+
+The grid track is the area between two adjacent row grid lines or adjacent column grid lines.
+
+### grid-area
+
+The grid area is the area surrounded by any four grid lines. A grid area is comprised of one or more grid cells.
+
+### display
+
+```css
+.grid-parent-container {
+  display: grid;
+}
+
+.inline-grid-parent-container {
+  display: inline-grid;
+}
+```
+
+Here, the `display: grid` property defines the parent grid container to be element with class 'grid-parent-container'. This generates a block level grid. This means that the grid takes up the entire line and other elements cannot co-exist with the grid.
+
+The `display: inline-grid` property defines the parent grid container to be element with class 'inline-grid-parent-container'. This generates an inline-level-grid. Other elemenets can take up the free space on the same line.
+
+### grid-template-rows & grid-template-columns
+
+These properties define how the rows and columns of the grid layout are arranged. They set the grid-track size.
+
+```html
+<style>
+  .grid-child {
+    background-color: #d0d1cd;
+  }
+  .grid-parent-container {
+    grid-template-columns: 200px 100px auto 100px 200px;
+    grid-template-rows: 25% 50% 25%;
+  }
+</style>
+<div class="grid-parent-container">
+  <div class="grid-child-1 grid-child">Grid Child 1</div>
+  <div class="grid-child-2 grid-child">Grid Child 2</div>
+  <div class="grid-child-3 grid-child">Grid Child 3</div>
+  <div class="grid-child-4 grid-child">Grid Child 4</div>
+  <div class="grid-child-5 grid-child">Grid Child 5</div>
+  <div class="grid-child-6 grid-child">Grid Child 6</div>
+  <div class="grid-child-7 grid-child">Grid Child 7</div>
+  <div class="grid-child-8 grid-child">Grid Child 8</div>
+  <div class="grid-child-9 grid-child">Grid Child 9</div>
+  <div class="grid-child-10 grid-child">Grid Child 10</div>
+  <div class="grid-child-11 grid-child">Grid Child 11</div>
+  <div class="grid-child-12 grid-child">Grid Child 12</div>
+  <div class="grid-child-13 grid-child">Grid Child 13</div>
+  <div class="grid-child-14 grid-child">Grid Child 14</div>
+  <div class="grid-child-15 grid-child">Grid Child 15</div>
+</div>
+```
+
+The above code creates a grid of 3 rows of 25%, 50% and 25% size of the parent container, and 5 columns of 100px, 50px, 50px and 100px. The middle column would automatically take up the rest of the space.
+
+![Grid Template Areas](/engineering-education/css-grid/grid-template.png)
+
+We can also use the `fr` unit. One fr or fraction takes up a fraction of the free space available in the grid container.
+
+![Grid Fraction](/engineering-education/css-grid/grid-fr.png)
+
+```html
+<style>
+  .grid-child {
+    background-color: #d0d1cd;
+  }
+
+  .grid-parent-container {
+    display: grid;
+    grid-template-columns: 1fr 1fr 1fr 1fr;
+    column-gap: 20px;
+    row-gap: 20px;
```

Comment:
While it's pretty self-explanatory, you could perhaps briefly mention the column-gap and row-gap properties you've used.